app/gui/src/project-view/components/GraphEditor/GraphNode.vue

```diff
@@ -232,14 +230,26 @@ const isVisualizationEnabled = computed({
     emit('update:visualizationEnabled', enabled)
   },
 })
+const visualizationHovered = ref(false)
+
 const isVisualizationPreviewed = computed(
-  () => keyboard.mod && outputHovered.value && !isVisualizationEnabled.value,
+  () =>
+    keyboard.mod &&
+    (outputHovered.value || visualizationHovered.value || nodeHovered.value) &&
+    !isVisualizationEnabled.value,
 )
 const isVisualizationVisible = computed(
   () => isVisualizationEnabled.value || isVisualizationPreviewed.value,
 )
+watch(isVisualizationVisible, (val) => {
+  if (!val) {
+    visualizationHovered.value = false
+  }
+})
```

Comment:
It's strange this is needed; hiding visualization itself should emit "pointerleave", effectively setting visualizationHovered to false. Is this needed due to the modifiers problem? If yes, then add a comment here.

Reply:
Yeah, I wasn’t paying enough attention here. We’re not getting pointerleave for whatever reason. I just added a comment about it.